src/components/main.vue

```diff
@@ -65,12 +65,15 @@ export default defineComponent({
       () => x.noResults && !x.semanticQueries.length && !relatedPrompts.value?.length,
     )
 
-    const isLowResult = computed(
-      () => x.totalResults > 0 && x.totalResults < semanticQueriesConfig.value.threshold,
+    const showSemantics = computed(
+      () =>
+        x.totalResults > 0 &&
+        x.totalResults < semanticQueriesConfig.value.threshold &&
+        !relatedPrompts.value?.length,
     )
 
     const showRelatedPrompts = computed(
-      () => (x.noResults || isLowResult.value) && relatedPrompts.value?.length,
+      () => (x.noResults || x.totalResults < 24) && relatedPrompts.value?.length,
```

Comment:
Hi! Instead of hardcoding the value, we can use the [config.pageSize](https://github.com/empathyco/x/blob/main/packages/x-components/src/x-modules/search/store/module.ts#L32) from the search x-module.